rest/route/host.go

```diff
@@ -163,21 +164,46 @@ func (h *hostModifyHandler) Run(ctx context.Context) gimlet.Responder {
 		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error for find() by distro id '%s'", h.hostID))
 	}
 
+	// Check if tags are valid
+	if err := validateTags(foundHost, h.AddInstanceTags, h.DeleteInstanceTags); err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "Invalid tag modifications"))
+	}
+
+	// Create new spawnhost modify job
 	changes := host.HostModifyOptions{
 		AddInstanceTags:    h.AddInstanceTags,
```

Comment:
Should you make sure that add CanBeModified = true? Somewhere anyway, since that's not something the user should be able to choose.